src/sentry/integrations/issues.py

```diff
@@ -96,6 +96,60 @@ def get_link_issue_config(self, group, **kwargs):
             }
         ]
 
+    def get_persisted_default_config_fields(self):
+        """
+        Returns a list of field names that should have their last used values
+        persisted on a per-project basis.
+        """
+        return []
+
+    def store_issue_last_defaults(self, project_id, data):
+        """
+        Stores the last used field defaults on a per-project basis. This
+        accepts a dict of values that will be filtered to keys returned by
+        ``get_persisted_default_config_fields`` which will automatically be
+        merged into the associated field config object as the default.
+
+        >>> integ.store_issue_last_defaults(1, {'externalProject': 2})
+
+        When the integration is serialized these values will automatically be
+        merged into the field configuration objects.
+
+        NOTE: These are currently stored for both link and create issue, no
+              differentiation is made between the two field configs.
+        """
+        persisted_fields = self.get_persisted_default_config_fields()
+        if not persisted_fields:
+            return
+
+        defaults = {k: v for k, v in six.iteritems(data) if k in persisted_fields}
+
+        self.org_integration.config.update({
+            'project_issue_defaults': {project_id: defaults},
+        })
+        self.org_integration.save()
+
+    def merge_issue_config_defaults(self, project_id, config):
+        # project_id is a long, but stored in the config as a string
+        project_id = six.text_type(project_id)
+        defaults = self.get_project_defaults(project_id)
+
+        if not defaults:
+            return config
+
+        updated_configs = [f.copy() for f in config]
```

Comment:
why do you need to copy these?

Reply:
i guess it's just to not mutate an argument?

Reply:
if so, at least a comment could be helpful. or you could also just mutate and make that clear from a docstring

Reply:
you could also do something like just call the `get_config` method from this method and pass in an `action` arg or something. a little gross but maybe makes more sense if you're not using the unaltered config outside of this method

Reply:
We copy since we mutate them after. There was a more specific reason I needed to copy them, and unfortunately I don't 100% remember now and should have definitely left a comment.